ledger/eval_simple_test.go

```diff
@@ -703,6 +705,105 @@ func TestAbsenteeChallenges(t *testing.T) {
+	ledgertesting.TestConsensusRange(t, 40, 0, func(t *testing.T, ver int, cv protocol.ConsensusVersion, cfg config.Local) {
+		dl := NewDoubleLedger(t, genBalances, cv, cfg)
+		defer dl.Close()
+
+		stib := dl.txn(&txntest.Txn{
+			Type:            "appl",
+			Sender:          addrs[0],
+			ApprovalProgram: getOnlineStake,
+		})
+		stakeChecker := stib.ApplicationID
+		require.NotZero(t, stakeChecker)
+
+		// have addrs[1] go online, though it won't be visible right away
+		dl.txn(&txntest.Txn{
+			Type:        "keyreg",
+			Sender:      addrs[1],
+			VotePK:      [32]byte{0xaa},
+			SelectionPK: [32]byte{0xbb},
+		})
+
+		one := basics.Address{0xaa, 0x11}
+		two := basics.Address{0xaa, 0x22}
+		three := basics.Address{0xaa, 0x33}
+
+		checkState := func(addr basics.Address, online bool, expected uint64, total uint64) {
+			t.Helper()
+			if !online {
+				require.Zero(t, expected)
+			}
+			stib = dl.txn(&txntest.Txn{
+				Type:          "appl",
+				Sender:        addr,
+				ApplicationID: stakeChecker,
+			})
+			logs := stib.ApplyData.EvalDelta.Logs
+			require.Len(t, logs, 3)
+			if online {
+				require.Equal(t, "\x00\x00\x00\x00\x00\x00\x00\x01", logs[0])
+				require.Equal(t, int(expected), int(binary.BigEndian.Uint64([]byte(logs[1]))))
+			} else {
+				require.Equal(t, "\x00\x00\x00\x00\x00\x00\x00\x00", logs[0])
+				require.Equal(t, int(expected), int(binary.BigEndian.Uint64([]byte(logs[1]))))
+			}
+			require.Equal(t, int(total), int(binary.BigEndian.Uint64([]byte(logs[2]))))
+		}
+
+		checkState(addrs[0], true, 833_333_333_333_333, 833_333_333_333_333)
+		// checking again because addrs[0] just paid a fee, but we show online balance hasn't changed yet
+		checkState(addrs[0], true, 833_333_333_333_333, 833_333_333_333_333)
+		for i := 1; i < 10; i++ {
+			checkState(addrs[i], false, 0, 833_333_333_333_333)
+		}
+
+		// Fund the new accounts and have them go online.
+		for i, addr := range []basics.Address{one, two, three} {
+			dl.txns(&txntest.Txn{
+				Type:     "pay",
+				Sender:   addrs[0],
+				Receiver: addr,
+				Amount:   (uint64(i) + 1) * 1_000_000_000,
+			}, &txntest.Txn{
+				Type:        "keyreg",
+				Sender:      addr,
+				VotePK:      [32]byte{byte(i + 1)},
+				SelectionPK: [32]byte{byte(i + 1)},
+			})
+		}
+		// they don't have online stake yet
+		for _, addr := range []basics.Address{one, two, three} {
+			checkState(addr, false, 0, 833_333_333_333_333)
+		}
+		for i := 0; i < 320; i++ {
+			dl.fullBlock()
+		}
+		// addr[1] is now visibly online. the total is across all five that are now online, minus various fees paid
+		checkState(addrs[1], true, 833_333_333_333_333-2000, 2*833_333_333_333_333-14000)
+		for i := 2; i < 10; i++ { // addrs[2-9] never came online
+			checkState(addrs[i], false, 0, 2*833_333_333_333_333-14000)
+		}
+		for i, addr := range []basics.Address{one, two, three} {
+			checkState(addr, true, (uint64(i)+1)*1_000_000_000-2000, 2*833_333_333_333_333-14000)
```

Comment:
I'd like to see test coverage of an account that was online, then got suspended. My understanding is `voter_params_get` should return no data for a suspended account.

Though this test might be easier to add to one of the incentives tests which already has access to a suspended account.

Reply:
You are a gentleman and a scholar.

Reply:
This should demonstrate it. 
https://github.com/algorand/go-algorand/pull/5984/commits/8e306a6ad9d54514a22b6eb73e0a850a7793c8ba
The test now goes forward 320 rounds and shows that `voter_params_get` starts returning nothing 320 rounds after suspension.